### DISCUSSION TITLE
How to sequence command execution?

### DISCUSSION THREAD

**Discussion by wigster** [humbertogontijo/homeassistant-roborock#305] [Q&A]
Hi,

I'd like to have a script which would first just vacuum a segment and then just mop this segment on a second pass, stringing the rooms together. It is possible to send a sequence of commands to the Roborock somehow and have it perform them as a sequence, instead of overriding the previous command? 

It is possible to send a list of segments together with a fixed parameter for fan speed etc, but here I'd like to be changing the mode of operation. The custom mop/fan mode partially takes care of this, since it changes the settings for each segment according to the app, but the mode for a particular room is fixed so I can't vacuum then mop.

The only way i can think of is to issue the vacuum command for one segment, then wait for the status to change to returning to dock and then issue the mop command. However, the status only gets updated when the vacuum is somewhere halfway back to the dock. Is there a better way?

**Comment by Lash-L** [comment]
Interesting question - once you figure it all out you should post it here so others can see. 

I'd recommend instead of docking, try just calling stop. That may be faster.

Then I haven't experimented with this service, but there is a service called update entity in the base home assistant services. If you use that it should theoretically call our update data function and updating the status of the vacuum earlier. 

But also thinking about it, I think the status is supposed to update after every time we send a command. Not sure but worth a shot

**Reply by wigster** [reply]
OK -- I am trying to put some logic together in script.

Would you know if there is a difference to any of the status attributes if the vacuum's finished its job vs it is interrupted because it needs to recharge and will return to finish later?
